integration-testing/build.gradle.kts

```diff
@@ -220,6 +221,47 @@ tasks {
         classpath = sourceSet.runtimeClasspath
     }
 
+    // In this test coroutine debug agent is attached as a javaagent vm argument
+    // to a pure Java project (safeDebugAgentTest) with no Kotlin stdlib dependency.
+    // In this case the error should be thrown from AgetnPremain class:
+    // "java.lang.IllegalStateException: kotlinx.coroutines debug agent failed to load."
+    val safeDebugAgentTest = create<Exec>("safeDebugAgentTest") {
+        // take kotlinx-coroutines-core.jar from a sourceSet with coroutines dependency
+        val sourceSet = sourceSets["coreAgentTest"]
+        val coroutinesDebugJar = sourceSet.runtimeClasspath.filter {
+            it.name == "kotlinx-coroutines-core-jvm-$coroutinesVersion.jar"
+        }.singleFile
+
+        commandLine(
+            if (System.getProperty("os.name").toLowerCase().contains("windows"))
+                "gradlew.bat"
+            else
+                "./gradlew",
+            ":safeDebugAgentTest:run",
+            "-Pdebug.agent=${coroutinesDebugJar.absolutePath}"
```

Comment:
`kotlinx.coroutines` is used in compiler integration tests, where additional parameters are passed to Gradle to indicate which compiler version to use, among other things. This command-line invocation can occasionally break in that scenario: imagine that `kotlinx.coroutines` uses `2.2.0`, but someone tests `kotlinx.coroutines` with `2.4.0-Beta`. The locally published `kotlinx-coroutines-core` JAR will be compiled by `2.4.0-Beta`, but this invocation will still use 2.2.0, which is a mismatch prohibited by the compiler (more than one major version).